ocaml/typing/layouts.ml

```diff
@@ -1015,8 +960,18 @@ module Layout = struct
   let equate = equate_or_equal ~allow_mutation:true
 
   let combine_histories reason lhs rhs =
-    Interact { reason; lhs_layout = lhs.layout; lhs_history = lhs.history;
-               rhs_layout = rhs.layout; rhs_history = rhs.history }
+    if flattened_histories
+    then begin match sub_desc (get_internal lhs.layout) (get_internal rhs.layout) with
+      | Sub -> lhs.history
+      | Not_sub -> rhs.history
+      | Equal -> begin match lhs.history, rhs.history with
+        (* Prefer other creation_reasons over Concrete_creation *)
+        | h, Creation (Concrete_creation _)
+        | Creation (Concrete_creation _), h -> h
+        | h, _ -> h
```

Comment:
Might be interesting to compare error messages with `h, _` vs `_, h`. Not sure what to expect there. Is one better than the other?

Reply:
Did a quick test and the difference between the two are quite minimal.

Going from left-biased to right-biased resulted in these two types of diff across the test cases:

```diff
--- a/ocaml/testsuite/tests/typing-layouts/modules_beta.ml
+++ b/ocaml/testsuite/tests/typing-layouts/modules_beta.ml
@@ -161,7 +161,7 @@ Error: This expression has type string but an expression was expected of type
        The layout of string is value, because
          it equals the primitive value type string.
        But the layout of string must be a sublayout of immediate, because
-         of the annotation on 'a in the declaration of the type t.
+         of the annotation on 'a in the declaration of the type s2'.
 |}]
```

```diff
--- a/ocaml/testsuite/tests/typing-layouts-float64/parsing.ml
+++ b/ocaml/testsuite/tests/typing-layouts-float64/parsing.ml
@@ -105,7 +105,7 @@ Error: This type float# should be an instance of type ('a : value)
        The layout of float# is float64, because
          it equals the primitive value type float#.
        But the layout of float# must be a sublayout of value, because
-         a term-level argument to a class constructor.
+         a type argument defaulted to have layout value.
 |}];;
```

The left-biased one looks to be a bit better.

Reply:
OK. Let's go with that then.